TodoApp.ConsoleApp/Repositories/XMLRepository/RepositoryBase.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Xml.Linq;
+using TodoApp.ConsoleApp.Repositories.Interfaces;
+using TodoApp.Library.Data;
+using System.Xml;
+using System.Linq;
+using System.IO;
+using System.Data;
+
+namespace TodoApp.ConsoleApp.Repositories.Models.XmlRepository
+{
+    abstract class RepositoryBase<TModel, TId> : IRepository<TModel, TId> where TModel : Imodel<TId>
+    {
+        protected string Path { get; }
+        protected XDocument Document { get; }
+        protected XElement ContainerElement { get; }
+
+        protected RepositoryBase(string path,XName containerName)
+        {
+            Path = path;
+            Document = XDocument.Load(path);
+            ContainerElement = Document.Element(containerName);
+        }
+
+        protected abstract TModel ElementToEntity(XElement element);
+        protected abstract XElement EntityToElement(TModel entity);
+
+        public IEnumerable<TModel> GetAll()
+        {
+            List<TModel> models = new List<TModel>();
+
+            foreach (var element in XMLContent())
+            {
+                models.Add(ElementToEntity(element));
+            }
+            return models;
+        }
+
+        public void Delete(TId id)
+        {
+            ContainerElement.Elements().FirstOrDefault(a => a.Attribute("id").Value == id.ToString()).Remove();
+        }
+
+        public IEnumerable<TModel> Get()
+        {
+            throw new NotImplementedException();
+        }
+
+        public TModel GetById(TId id)
+        {
+            XElement element = ContainerElement.Elements().FirstOrDefault(a => a.Attribute("id").Value == id.ToString());
+
+            TModel model = ElementToEntity(element);
+
+            return model;
+        }
+
+        public TModel Insert(TModel model)
+        {
+            Guid id = new Guid();
+            XElement element = EntityToElement(model);
+            element.Attribute("id").SetValue(id);
+            TModel modelwithId = ElementToEntity(element);
+
+            return modelwithId;
```

Comment:
Looks good 👍 
Just some minor changes:
- rename variables so their purpose is obvious
- the declaration of variables must be as closer as possible to where there are used (in this case `id`)
```suggestion
            XElement newElement = EntityToElement(model);
            
            Guid id = new Guid();
            newElement.Attribute(IdName).SetValue(id);
            
            TModel newModel = ElementToEntity(element);
            return newModel;
```